File: Tyche.Android/data/bet/src/main/java/com/felipearpa/tyche/data/bet/domain/PoolGamblerBetResponse.kt
package com.felipearpa.tyche.data.bet.domain

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
internal data class PoolGamblerBetResponse(
    val poolId: String,
    val gamblerId: String,
    val matchId: String,
    val homeTeamId: String,
    val homeTeamName: String,
    val homeTeamScore: Int?,
    val homeTeamBet: Int?,
    val awayTeamId: String,
    val awayTeamName: String,
    val awayTeamScore: Int?,
    val awayTeamBet: Int?,
    val score: Int?,
    @Contextual val matchDateTime: LocalDateTime,
    val isLocked: Boolean,
)

File: Amazon.Lambda/Felipearpa.Tyche.Function/src/Response/PoolGamblerBetResponse.fs
namespace Felipearpa.Tyche.Function.Response

open System

type PoolGamblerBetResponse =
    { PoolId: string
      GamblerId: string
      MatchId: string
      HomeTeamId: string
      HomeTeamName: string
      HomeTeamScore: int option
      HomeTeamBet: int option
      AwayTeamId: string
      AwayTeamName: string
      AwayTeamScore: int option
      AwayTeamBet: int option
      Score: int option
      MatchDateTime: DateTime
      isLocked: bool }
